rotkehlchen/data/importer.py

```diff
@@ -1435,3 +1446,94 @@ def import_bisq_trades_csv(self, filepath: Path, **kwargs: Any) -> Tuple[bool, s
                 except KeyError as e:
                     return False, str(e)
         return True, ''
+
+    @staticmethod
+    def _group_binance_rows(
+        rows: List[BinanceCsvRow],
+        **kwargs: Dict[str, Any],
+    ) -> Tuple[int, Dict[Timestamp, List[BinanceCsvRow]]]:
+        multirows: Dict[Timestamp, List[BinanceCsvRow]] = defaultdict(lambda: [])
+        skipped_count = 0
+        formatstr = kwargs.get('timestamp_format')
+        for csv_row in rows:
+            try:
+                timestamp = Timestamp(deserialize_timestamp_from_date(
+                    date=csv_row['UTC_Time'],
+                    formatstr=formatstr if formatstr is not None else '%Y-%m-%d %H:%M:%S',  # type: ignore  # noqa: E501
+                    location='binance',
+                ))
+                csv_row['Coin'] = asset_from_binance(csv_row['Coin'])
+                csv_row['Change'] = deserialize_asset_amount(csv_row['Change'])
+                del csv_row['UTC_Time']
+                del csv_row['User_ID']
+                del csv_row['Account']
+                del csv_row['Remark']
+                multirows[timestamp].append(csv_row)
+            except (DeserializationError, UnknownAsset) as e:
+                log.warning(f'Skipped binance csv row {csv_row} because of {str(e)}')
+                skipped_count += 1
+            except KeyError as e:
+                log.error(f'Malformed binance csv columns! Broke on row {csv_row}. {str(e)}')
+                return len(rows), {}
+
+        return skipped_count, multirows
+
+    def _process_binance_rows(self, multi: Dict[Timestamp, List[BinanceCsvRow]]) -> None:
+        BinanceEntry.db = self.db
+        BinanceEntry.db_ledger = self.db_ledger
+        single_entry_classes = BinanceSingleEntry.__subclasses__()
+        multiple_entry_classes = BinanceMultipleEntry.__subclasses__()
+        stats: Dict[Type[BinanceEntry], int] = defaultdict(int)
+        skipped_entries: List[Any] = []
+        for timestamp, rows in multi.items():
+            for single_entry_class in single_entry_classes:
+                for row in rows.copy():
```

Comment:
Why do you need to copy it here? I'm checking and `process_entry` doesn't modify the row + you are not keeping the changes anywhere. I believe the `.copy` can be removed

Reply:
I modify `rows` at the end of the loop on the line 1494 (`rows.remove(row)`) so used copy there. If `copy` is not good, do you have any suggestions on how to replace it?

Reply:
ah okey, you use copy to not modify the same list you are iterating... I believe copy is good for performance so should be okey